### PR TITLE
check that public API of dart:ui conforms to web implementation

### PR DESCRIPTION
**Pull request by jonahwilliams**
This currently only checks public methods of public classes. We don't yet verify statics or top levels. There is no type checking.

### CODE REVIEW

File: web_sdk/test/api_conform_test.dart [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  bool failed = false;
+  print('Checking ${uiClasses.length} public classes.');
+  for (String className in uiClasses.keys) {
+    final ClassDeclaration uiClass = uiClasses[className];
+    final ClassDeclaration webClass = webClasses[className];
+    // If the web class is missing there isn't much left to do here. Print a
+    // warning and move along.
+    if (webClass == null) {
+      failed = true;
+      print('Warning: io/dart:ui contained public class $className, but '
```
Comment (goderbauer) [inline web_sdk/test/api_conform_test.dart]: Could all these warning be made more actionable? For example, we could call out that people can find io/dart:ui in 'lib/ui/' and the other one in 'lib/stub_ui/' to make it clear where things need to be added when this fails.

File: web_sdk/test/api_conform_test.dart [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    _collectPublicMethods(uiClass, uiMethods);
+    _collectPublicMethods(webClass, webMethods);
+
+    for (String methodName in uiMethods.keys) {
+      final MethodDeclaration uiMethod = uiMethods[methodName];
+      final MethodDeclaration webMethod = webMethods[methodName];
+      if (webMethod == null) {
+        failed = true;
+        print(
+          'Warning: io/dart:ui $className.$methodName is missing from web/dart:ui',
```
Comment (goderbauer) [inline web_sdk/test/api_conform_test.dart]: Style nit: the first warning ended with a `.`, the others don't.

**Comment by jonahwilliams** [comment]
Why can't I write a bash script? 🥇

File: web_sdk/test/api_conform_test.dart (hunk 1 above)
Comment (liyuqian) [reply web_sdk/test/api_conform_test.dart]: The new error message still doesn't look very actionable to me... Should I interpret it as "Please copy the missing classes and functions from lib/ui/ui.dart to lib/stub_ui/ui.dart."?

**Pull request merged**